### DISCUSSION TITLE
Sentence Bert

### DISCUSSION THREAD

**Discussion by lxmm0427** [jishnujayakumar/MLRC2020-EmbedKGQA#16] [Q&A]
Hi, 
You mentioned [here](https://github.com/malllabiisc/EmbedKGQA/issues/58) that SBert performed better than Roberta.I tried a few times but couldn't get better results.
Are the hyperparameters the same as Roberta when using SBert? I would appreciate it if you could provide hyperparameters that you used to get a better result.

**Comment by jishnujayakumar** [comment]
Hi @lxmm1999, kindly try using the following setting and post your experience.

```bash
##### WebQSP
--relation_dim 200 
--que_embedding_model SentenceTransformer 
--do_batch_norm 0
--freeze 1 
--batch_size 16 
--validate_every 10  
--lr 0.00002 
--entdrop 0.0 
--reldrop 0.0 
--scoredrop 0.0
--decay 1.0 
--model ComplEx 
--patience 20 
--ls 0.0 
--l3_reg 0.001 
--nb_epochs 200
```

**Reply by lxmm0427** [reply]
using command
`python3 main.py --mode train --hops webqsp_half --relation_dim 200 --que_embedding_model SentenceTransformer --do_batch_norm 0 --gpu 1 --freeze 1 --batch_size 16 --validate_every 10 --lr 0.00002 --entdrop 0.0 --reldrop 0.0 --scoredrop 0.0 --decay 1.0 --model ComplEx --patience 20 --ls 0.0 --l3_reg 0.001 --nb_epochs 200`
I got :

> ComplEx,half,SentenceTransformer,0.43932511356262166,0.43932511356262166,0.5373134328358209,0.5554834523036989

but using RoBERTa
`python3 main.py --mode train --relation_dim 200 --do_batch_norm 0 --gpu 3 --freeze 1 --batch_size 32 --validate_every 10 --hops webqsp_half --lr 0.00002 --entdrop 0.3 --reldrop 0.2 --scoredrop 0.2 --decay 1.0 --model ComplEx --patience 20 --ls 0.0 --l3_reg 0.0 --nb_epochs 200`
I got 
> ComplEx,half,RoBERTa,0.47306943543153795,0.47306943543153795,0.5561323815704088,0.5807916937053861

It seems that RoBERTa performs better

**Reply by jishnujayakumar** [reply]
In order to compare two things, kindly experiment with the same set of parameters rather than different. If you have any further doubts, kindly contact the original author for the original set of hyperparameters.
